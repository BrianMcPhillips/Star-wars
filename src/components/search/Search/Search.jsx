import React from 'react';
import PropTypes from 'prop-types';
import { terms } from '../../../assets/data';

const Search = ({
  searchBy,
  searchTerm,
  handleSearchBy,
  handleSearchTerm,
  makeRequest
}) => (
  <div>
    <h3>Search</h3>
    <input onChange={handleSearchTerm} type="text" value={searchTerm}/>
    <select onChange={handleSearchBy} value={searchBy}>
      {
        terms.map((option, i) => 
          <option key={option + i} value={option}>{option}</option>
        )
      }
    </select>
    <button onClick={makeRequest}>Search</button>
  </div>
);

Search.propTypes = {
  searchBy: PropTypes.string.isRequired,
  searchTerm: PropTypes.string.isRequired,
  handleSearchBy: PropTypes.func.isRequired,
  handleSearchTerm: PropTypes.func.isRequired,
  makeRequest: PropTypes.func.isRequired
};

export default Search;
