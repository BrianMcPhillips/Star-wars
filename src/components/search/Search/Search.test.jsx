import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from './Search';

describe('Search component', () => {
  afterEach(() => cleanup());
  it('renders Search', () => {
    const { asFragment } = render(
      <Search 
        searchBy="people"
        searchTerm="R2"
        handleSearchBy={() => {}}
        handleSearchTerm={() => {}}
        makeRequest={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
