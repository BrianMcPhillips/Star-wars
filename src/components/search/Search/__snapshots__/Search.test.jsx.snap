// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Search
    </h3>
    <input
      type="text"
      value="R2"
    />
    <select>
      <option
        value="films"
      >
        films
      </option>
      <option
        value="people"
      >
        people
      </option>
      <option
        value="planets"
      >
        planets
      </option>
      <option
        value="species"
      >
        species
      </option>
      <option
        value="starships"
      >
        starships
      </option>
      <option
        value="vehicles"
      >
        vehicles
      </option>
    </select>
    <button>
      Search
    </button>
  </div>
</DocumentFragment>
`;
